client/asset/dcr/spv.go

```diff
@@ -0,0 +1,1020 @@
+		return nil, err
+	}
+
+	var addrStr string
+	for _, detail := range details {
+		if detail.Vout == index {
+			addrStr = detail.Address
+		}
+	}
+	if addrStr == "" {
+		return nil, fmt.Errorf("error locating address for output")
+	}
+
+	tree := wire.TxTreeRegular
+	if txd.TxType != stake.TxTypeRegular {
+		tree = wire.TxTreeStake
+	}
+
+	if len(txd.MsgTx.TxOut) <= int(index) {
+		return nil, fmt.Errorf("not enough outputs")
+	}
+
+	_, tipHeight := w.MainChainTip(ctx)
+
+	var ours bool
+	for _, credit := range txd.Credits {
+		if credit.Index == index {
+			if credit.Spent {
+				return nil, asset.CoinNotFoundError
+			}
+			ours = true
+			break
+		}
+	}
+
+	if !ours {
+		return nil, asset.CoinNotFoundError
+	}
+
+	return &TxOutput{
+		TxOut:         txd.MsgTx.TxOut[index],
+		Tree:          tree,
+		Addresses:     []string{addrStr},
+		Confirmations: uint32(txd.Block.Height - tipHeight + 1),
+	}, nil
+}
+
+// ExternalAddress returns an external address using GapPolicyIgnore.
+// Part of the Wallet interface.
+//
+// NOTE: Why GapPolicyIgnore instead of GapPolicyWrap? Can we recover from an
+// ignored gap policy without extreme measures?
+func (w *spvWallet) ExternalAddress(ctx context.Context, _ string) (stdaddr.Address, error) {
+	return w.NewExternalAddress(ctx, w.acctNum, wallet.WithGapPolicyIgnore())
+}
+
+// InternalAddress returns an internal address using GapPolicyIgnore.
+// Part of the Wallet interface.
+func (w *spvWallet) InternalAddress(ctx context.Context, _ string) (stdaddr.Address, error) {
+	return w.NewInternalAddress(ctx, w.acctNum, wallet.WithGapPolicyIgnore())
+}
+
+// SignRawTransaction signs the provided transaction.
+// Part of the Wallet interface.
+func (w *spvWallet) SignRawTransaction(ctx context.Context, baseTx *wire.MsgTx) (*wire.MsgTx, error) {
+	tx := baseTx.Copy()
+	sigErrs, err := w.dcrWallet.SignTransaction(ctx, tx, txscript.SigHashAll, nil, nil, nil)
+	if err != nil {
+		return nil, err
+	}
+	if len(sigErrs) > 0 {
+		for _, sigErr := range sigErrs {
+			w.log.Errorf("signature error for index %d: %v", sigErr.InputIndex, sigErr.Error)
+		}
+		return nil, fmt.Errorf("%d signature errors", len(sigErrs))
+	}
+	return tx, nil
+}
+
+// SendRawTransaction broadcasts the provided transaction to the Decred network.
+// Part of the Wallet interface.
+func (w *spvWallet) SendRawTransaction(ctx context.Context, tx *wire.MsgTx, allowHighFees bool) (*chainhash.Hash, error) {
+	// TODO: Conditional high fee check?
+	return w.PublishTransaction(ctx, tx, w.spv)
+}
+
+// GetBlockHeader generates a *BlockHeader for the specified block hash. The
+// returned block header is a wire.BlockHeader with the addition of the block's
+// median time and other auxiliary information.
+func (w *spvWallet) GetBlockHeader(ctx context.Context, blockHash *chainhash.Hash) (*BlockHeader, error) {
+	hdr, err := w.dcrWallet.BlockHeader(ctx, blockHash)
+	if err != nil {
+		return nil, err
+	}
+
+	medianTime, err := w.medianTime(ctx, blockHash)
```

Comment:
The first thing that `medianTime` does is `w.dcrWallet.BlockHeader(ctx, blockHash)`.  Might consider having `medianTime` return that header although it's a minor redundancy.